op-proposer/proposer/driver.go

```diff
@@ -303,18 +305,52 @@ func (l *L2OutputSubmitter) FetchOutput(ctx context.Context, block *big.Int) (*e
 }
 
 // ProposeL2OutputTxData creates the transaction data for the ProposeL2Output function
-func (l *L2OutputSubmitter) ProposeL2OutputTxData(output *eth.OutputResponse) ([]byte, error) {
-	return proposeL2OutputTxData(l.l2ooABI, output)
+func (l *L2OutputSubmitter) ProposeL2OutputTxData(output *eth.OutputResponse, consumerId string) ([]byte, error) {
+	// fetch pub rand and finality signature from the EOTS aggregator
+	res, err := l.EotsAggClient.GetEOTSInfos(int64(output.BlockRef.Number), consumerId)
+	if err != nil {
+		return nil, err
+	}
+	var eotsInfos []bindings.TypesEOTSInfo
+	for _, ptr := range res {
+		if ptr != nil {
+			item := bindings.TypesEOTSInfo{
+				FpBtcPk:     ptr.FinalitySig,
+				PubRand:     ptr.PubRand,
+				FinalitySig: ptr.FinalitySig,
+			}
+			eotsInfos = append(eotsInfos, item)
+		}
+	}
+	return proposeL2OutputTxData(l.l2ooABI, output, eotsInfos)
 }
 
 // proposeL2OutputTxData creates the transaction data for the ProposeL2Output function
-func proposeL2OutputTxData(abi *abi.ABI, output *eth.OutputResponse) ([]byte, error) {
+func proposeL2OutputTxData(abi *abi.ABI, output *eth.OutputResponse, eotsInfos []bindings.TypesEOTSInfo) ([]byte, error) {
 	return abi.Pack(
 		"proposeL2Output",
 		output.OutputRoot,
 		new(big.Int).SetUint64(output.BlockRef.Number),
 		output.Status.CurrentL1.Hash,
-		new(big.Int).SetUint64(output.Status.CurrentL1.Number))
+		new(big.Int).SetUint64(output.Status.CurrentL1.Number), eotsInfos)
+}
+
+// ConstructConsumerID constructs a consumer ID
+// following the convention 'op-stack-l2-<L2 ChainID>'.
+func (l *L2OutputSubmitter) ConstructConsumerID(ctx context.Context) (string, error) {
+	cCtx, cancel := context.WithTimeout(ctx, l.Cfg.NetworkTimeout)
```

Comment:
what's NetworkTimeout?

Reply:
It is a Timeout config for network operation and it is init with default value [10s].(https://github.com/babylonchain/optimism/blob/develop-babylon/op-service/txmgr/cli.go#L79)